### ISSUE TITLE
UI Function suggestion: "Lock Screen by default"

### ISSUE THREAD

**Issue by hiuwo** [debauchee/barrier#672]
Hi I was wondering if you could add a new UI for a function in the advanced server settings in the next release which is "Lock Cursor to Screen by default" to make mouse crossing screens unavailable on server start so that one can only do so by using hotkeys?

I know this is probably a side project but this would be amazing!

**Comment by simons-public** [comment]
@hiuwo If you're trying to avoid accidental switching there's a couple options in the Advanced Server Settings section under Server Configuration that can help.

There's "Switch after waiting" which delays switching by X milliseconds and "Switch on double tap within" which will only switch if you double-tap the edge of the screen in X milliseconds.

**Comment by hiuwo** [comment]
Thank you for the reply! Those are some great suggestions. I was only asking for Lock screen by default because it was already a built-in function/ command. Maybe most people don't realise it but if you go:

Configure Server -> Hotkeys (tab) -> New (Hotkey) -> New (Actions):

You can actually access the option there (along with some more useful ones) called "Lock Cursor to Screen". I currently just assign a key and use it habitually everytime I start up my computer but I was hoping I could make a small suggestion.👍

No biggie though, I am really appreciating  Barrier for existing, it just solved so many problems Synergy has.

**Comment by ThinkSalat** [comment]
Agreed. I use only one monitor but switch between pc and work laptop. There should be an option to lock by default or disable edge switching.

**Comment by nikatjef** [comment]
Greetings,

  I beg you, please do not enable this by default.  Make it an option that people need to actively choose.

**Comment by hiuwo** [comment]
yes a checkbox for such function to be set as default for subsequent sessions is what I am proposing. Everyone agrees with you, it would be nonsense to not add an option in the UI but just flip the program completely the opposite. Don’t worry.

**Comment by p12tic** [comment]
Let's not close valid feature requests.
